R/tar_insuvimeh.R

```diff
@@ -0,0 +1,123 @@
+      
+      fp_base <- basename(fp_nc)
+      # meta data collection  - to write band_name as "{pub_date}.lt_{leadtime}"
+      valid_time_meta <- ncmeta::nc_att(fp_nc,"T","units")$value$units
+      start_time_meta <- ncmeta::nc_att(fp_nc,"S","units")$value$units
+      valid_date <- as_date(str_extract(valid_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      start_date <- as_date(str_extract(start_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      lead_time_diff <- interval(start_date,valid_date)
+      lt <- lead_time_diff %/% months(1)
+      tibble(
+        file_name_parent = fp_par,
+        file_name= fp_base,
+        pub_date = start_date,
+        valid_date =valid_date,
+        lt = lt
+      )
+    }) %>% 
+    list_rbind()
+  return(lr)
+}
+
+
+load_insuvimeh_raster <-  function(gdb){
+  fps <- list.files(gdb,
+                    full.names =T,
+                    recursive = T,
+                    pattern = "\\d{4}.nc$")
+  
+  lr <- fps %>% 
+    map(\(fp_nc){
+      
+      # meta data collection  - to write band_name as "{pub_date}.lt_{leadtime}"
+      valid_time_meta <- ncmeta::nc_att(fp_nc,"T","units")$value$units
+      start_time_meta <- ncmeta::nc_att(fp_nc,"S","units")$value$units
+      valid_date <- as_date(str_extract(valid_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      start_date <- as_date(str_extract(start_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      lead_time_diff <- interval(start_date,valid_date)
+      lt <- lead_time_diff %/% months(1)
+      bname<- paste0(start_date,".lt_",lt)
+      
+      # Now we open the each file and turn it into a terra::rast() 
+      cat(bname,"\n")
+      dat <- ncdf4::nc_open(fp_nc)
+      lon <- ncdf4::ncvar_get(dat,"X")
+      lat <- ncdf4::ncvar_get(dat,"Y")
+      value_array <- ncdf4::ncvar_get(dat,"deterministic")
+      value_array_fixed <- aperm(value_array,c(2,1))
+      
+      rtmp <- terra::rast(
+        x= value_array_fixed,
+        extent =ext(
+          min(lon),
+          max(lon),
+          min(lat),
+          max(lat)
+        ),
+        crs= "OGC:CRS84"
+      )
+      # set correct band nanme
+      rtmp %>% 
+        set.names(bname)
+      return(rtmp)
+    }
+    )
+  # merge multi-band
+  r <- rast(lr ) 
+  
+  # wrap so it can be saved as a target.
+  r_wrap <- wrap(r)
+  return(r_wrap)
+  
+}
+
+
+
+
+
+#' zonal_gtm_insuvimeh
+#' @description
+#' custom targets function to run zonal means by leadtime and publication date for forecast provided by insivumeh
+#'
+#' @param r `packedSpatRaster` terra object created in pipeline
+#' @param gdf sf class polygon to run zonal statistics to
+#' @param rm_dup_years `logical` just parameter used in beginning when catalogue had issues
+#'
+#' @return `data.frame` in long format w/ zonal means by publication date and leadtime
+zonal_gtm_insuvimeh <-  function(r, gdf,rm_dup_years=T){
+  r_gtm <- unwrap(r)
+  if(rm_dup_years){
+    r_nm_yr_gtm <- as_date(str_extract(names(r_gtm),"\\d{4}-\\d{2}-\\d{2}") )
```

Comment:
Seems you do this parsing in the previous function as well, any way to just do it once?

Reply:
yeah previous function actually creates convenient band names for zonal extraction, in format: "YYYY-MM-DD.lt_int", this was just an intermediate work-around i needed to do when I found issues/duplication in the data provided- by converting band names back to date i could easily filter/manipulate the issue. 

It's actually set to `F` in the pipeline now that the data issue is fixed so this issue I kind of want to leave it in for a little while longer as a reminder, but maybe eventually delete... just changed the default value to F (same as pipeline) [6b7f0a6](https://github.com/OCHA-DAP/ds-aa-lac-dry-corridor/pull/12/commits/6b7f0a6a6a1a48240a5315771bf5eedf56c79550)